main.tf

```diff
@@ -229,9 +247,21 @@ resource "aws_cloudfront_distribution" "default" {
       domain_name = origin.value.domain_name
       origin_id   = origin.value.origin_id
       origin_path = lookup(origin.value, "origin_path", "")
+
+      dynamic "custom_header" {
```

Comment:
How is the `custom_header` above different from the `var.custom_origins` `origin` block `custom_header` ?

How come the same `custom_header` isn't applied to the `var.s3_origins` `origin` block below ?

Reply:
Good eye there, @nitrocode. The two `custom_header` sections must be merged. I have a solution written, but I must first upgrade Terraform in order to test.

Reply:
Tested the merged `custom_header` sections with Terraform 0.13 and 0.14.